### PR TITLE
[portsmf] add new port with version 236

### PR DESCRIPTION
**Pull request by Be-ing**
**Describe the pull request**

- #### What does your PR fix?  
Adds new port for [PortSMF](https://github.com/tenacityteam/portsmf)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all except UWP

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

File: ports/portsmf/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    SHA512 de2abaa4ffa695d51ce1882744c7625380297a294e0cf7042ff616b64abdedd149b94e8618d049e518d9c3611cfcb16dec8acfa302e2943d7dcb68e487d97731
+    HEAD_REF main
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+)
+
+vcpkg_install_cmake()
```
Comment (JonLiu1993) [inline ports/portsmf/portfile.cmake:16]:
```suggestion
vcpkg_cmake_install()
```

File: ports/portsmf/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+vcpkg_copy_pdbs()
+vcpkg_fixup_pkgconfig()
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/PortSMF TARGET_PATH share/PortSMF)
```
Comment (JonLiu1993) [inline ports/portsmf/portfile.cmake:19]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/PortSMF)
```

File: ports/portsmf/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+file(INSTALL ${SOURCE_PATH}/license.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (JonLiu1993) [inline ports/portsmf/portfile.cmake:21]:
```suggestion
file(INSTALL "${SOURCE_PATH}/license.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: ports/portsmf/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "portsmf",
+  "version-string": "0.235.1",
```
Comment (JonLiu1993) [inline ports/portsmf/vcpkg.json:3]:
```suggestion
  "version": "0.235.1",
```

File: ports/portsmf/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO tenacityteam/portsmf
+    REF 235.1
+    SHA512 de2abaa4ffa695d51ce1882744c7625380297a294e0cf7042ff616b64abdedd149b94e8618d049e518d9c3611cfcb16dec8acfa302e2943d7dcb68e487d97731
+    HEAD_REF main
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/portsmf/portfile.cmake:13]:
```suggestion

```

File: ports/portsmf/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO tenacityteam/portsmf
+    REF 235.1
+    SHA512 de2abaa4ffa695d51ce1882744c7625380297a294e0cf7042ff616b64abdedd149b94e8618d049e518d9c3611cfcb16dec8acfa302e2943d7dcb68e487d97731
+    HEAD_REF main
+)
+
+vcpkg_configure_cmake(
```
Comment (JonLiu1993) [inline ports/portsmf/portfile.cmake:11]:
```suggestion
vcpkg_cmake_configure(
```

File: ports/portsmf/vcpkg.json [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+  "description": "Portsmf is 'Port Standard MIDI File', a cross-platform, C++ library for reading and writing Standard MIDI Files.",
+  "homepage": "https://github.com/tenacityteam/portsmf",
+  "license": "MIT",
+  "supports": "!uwp"
```
Comment (JonLiu1993) [inline ports/portsmf/vcpkg.json:7]:
```suggestion
  "supports": "!uwp"
  "dependencies":[
  {
    "name": "vcpkg-cmake",
    "host": true
  },
  {
    "name": "vcpkg-cmake-config",
    "host": true
  }
  ]
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@Be-ing ，Could you modify the place suggested above

**Comment by Be-ing** [comment]
I'm confused why this is acceptable but #18978 is not?

**Comment by JonLiu1993** [comment]
> I'm confused why this is acceptable but #18978 is not?

Generally speaking, if the user updates the port version, we encourage to update the latest cmake function, but if it is to add a new port, we will let the user directly use the latest cmake function

**Comment by Be-ing** [comment]
That's not related to why #18978 was rejected.

**Comment by JonLiu1993** [comment]
> That's not related to why #18978 was rejected.

Sorry,for some major changes to the port, we must get official approval.

File: versions/baseline.json [hunk 8]
```diff
@@ -3120,14 +3120,14 @@
       "baseline": "0.6.0-1",
       "port-version": 0
     },
-    "libebur128": {
-      "baseline": "1.2.6",
-      "port-version": 0
-    },
```
Comment (JonLiu1993) [inline versions/baseline.json:3126]: @Be-ing , Could you clarify why delete and re-add?

**Review by JonLiu1993 — APPROVED** [review]

File: versions/baseline.json (hunk 8 above)
Comment (Be-ing) [reply versions/baseline.json:3126]: I don't know. I did not make any manual changes to this. I only ran the `vcpkg x-add-version` tool.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**